File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedChampionsUpdate.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.RankedChampion;

import java.util.List;

@CommandType("ranked champions updated")
public record RankedChampionsUpdate(
	List<RankedChampion> champions,
	@Json(name = "serieId")
	int seriesId
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/RankedChampionsUpdate.kt
package tech.zolhungaj.amqapi.servercommands.globalstate

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.RankedChampion

@JvmRecord
@CommandType("ranked champions updated")
data class RankedChampionsUpdate(
    val champions: List<RankedChampion>,
    @Json(name = "serieId") val seriesId: Int
)